cmd/guacone/cmd/analyze.go

```diff
@@ -0,0 +1,476 @@
+
+	var hasSBOMResponseOne *model.HasSBOMsResponse
+	var hasSBOMResponseTwo *model.HasSBOMsResponse
+	var err error
+	logger := logging.FromContext(ctx)
+
+	if opts.URI {
+		hasSBOMResponseOne, err = analyzer.FindHasSBOMBy(model.HasSBOMSpec{}, sboms[0], "", "", ctx, gqlclient)
+		if err != nil {
+			return []graph.Graph[string, *analyzer.Node]{}, fmt.Errorf("(uri)failed to lookup sbom: %v %v", sboms[0], err)
+		}
+		hasSBOMResponseTwo, err = analyzer.FindHasSBOMBy(model.HasSBOMSpec{}, sboms[1], "", "", ctx, gqlclient)
+		if err != nil {
+			return []graph.Graph[string, *analyzer.Node]{}, fmt.Errorf("(uri)failed to lookup sbom: %v %v", sboms[1], err)
+		}
+	} else if opts.PURL {
+		hasSBOMResponseOne, err = analyzer.FindHasSBOMBy(model.HasSBOMSpec{}, "", sboms[0], "", ctx, gqlclient)
+		if err != nil {
+			return []graph.Graph[string, *analyzer.Node]{}, fmt.Errorf("(purl)failed to lookup sbom: %v %v", sboms[0], err)
+		}
+		hasSBOMResponseTwo, err = analyzer.FindHasSBOMBy(model.HasSBOMSpec{}, "", sboms[1], "", ctx, gqlclient)
+		if err != nil {
+			return []graph.Graph[string, *analyzer.Node]{}, fmt.Errorf("(purl)failed to lookup sbom: %v %v", sboms[1], err)
+		}
+	} else if opts.ID {
+		hasSBOMResponseOne, err = analyzer.FindHasSBOMBy(model.HasSBOMSpec{}, "", "", sboms[0], ctx, gqlclient)
+		if err != nil {
+			return []graph.Graph[string, *analyzer.Node]{}, fmt.Errorf("(id)failed to lookup sbom: %v %v", sboms[0], err)
+		}
+		hasSBOMResponseTwo, err = analyzer.FindHasSBOMBy(model.HasSBOMSpec{}, "", "", sboms[1], ctx, gqlclient)
+		if err != nil {
+			return []graph.Graph[string, *analyzer.Node]{}, fmt.Errorf("(id)failed to lookup sbom: %v %v", sboms[1], err)
+		}
+	}
+
+	if hasSBOMResponseOne == nil || hasSBOMResponseTwo == nil {
+		return []graph.Graph[string, *analyzer.Node]{}, fmt.Errorf("failed to lookup sboms: nil")
+	}
+
+	if len(hasSBOMResponseOne.HasSBOM) == 0 || len(hasSBOMResponseTwo.HasSBOM) == 0 {
+		return []graph.Graph[string, *analyzer.Node]{}, fmt.Errorf("failed to lookup sboms, one endpoint may not have sboms")
+	}
+
+	if len(hasSBOMResponseOne.HasSBOM) != 1 || len(hasSBOMResponseTwo.HasSBOM) != 1 {
+		logger.Infof("Multiple sboms found for given purl, id or uri. Using first one")
+	}
+	hasSBOMOne := hasSBOMResponseOne.HasSBOM[0]
+	hasSBOMTwo := hasSBOMResponseTwo.HasSBOM[0]
+
+	//create graphs
+	gOne, err := analyzer.MakeGraph(hasSBOMOne, opts.Metadata, opts.InclSoft, opts.InclDeps, opts.InclOccur, opts.Namespaces)
+	if err != nil {
+		logger.Fatalf(err.Error())
+	}
+	gTwo, err := analyzer.MakeGraph(hasSBOMTwo, opts.Metadata, opts.InclSoft, opts.InclDeps, opts.InclOccur, opts.Namespaces)
+	if err != nil {
+		logger.Fatalf(err.Error())
+	}
+	return []graph.Graph[string, *analyzer.Node]{
+		gOne,
+		gTwo,
+	}, nil
+}
+
+func verifyAnalyzeFlags(slsas, sboms []string, errSlsa, errSbom error, uri, purl, id bool) error {
```

Comment:
nit: validate instead of verify for consistency https://github.com/guacsec/guac/blob/main/cmd/guaccollect/cmd/scorecard.go#L124